straxen/plugins/veto_pulse_processing.py

```diff
@@ -72,6 +83,46 @@ def compute(self, raw_records_coin_nv):
         rlinks = strax.record_links(r)
         r = clean_up_empty_records(r, rlinks, only_last=True)
         return r
+    
+    
+def median_baseline(records):
+    """
+    Function which computes the baseline according the pulse's median.
+    
+    :param records: Records 
+    """
+    # Count number of pulses and create buffer:
+    npulses = np.sum(records['record_i'] == 0)
+    
+    if npulses == 1:
+        # This case is simple
+        records = _correct_baseline(records)
+    else:
+        # Now the more complicated case in which we have multiple pulses:
+        _, nextr = strax.record_links(records)
+
+        pulse_i = []
+        for i in np.where(records['record_i'] == 0)[0]:
+            inds = [i]
+            ind = nextr[i]
+            while ind != -1:
+                inds += [ind]
+                ind = nextr[ind]
+            pulse_i.append(inds)
+
+        for pi in pulse_i:
+            records[pi] = _correct_baseline(records[pi])
+    return records
+
+
+def _correct_baseline(records):
```

Comment:
Can we numba-fy?

Reply:
See comment above

Reply:
Okay, numbafication brought a speed up of ~x1.5 - x2 so I added it (from 1.4 ms -> 800µs per saturation event). I also checked if the  `median_baseline` could become a bottleneck, but even if we would have 1k saturated events per run (which would indicate some problem). The processing only takes 700 ms.